doc/continuous_integration/dockerhub.md

```diff
@@ -0,0 +1,64 @@
+
+The build hooks, e.g. [build](/.dockerhub/debug/hooks/build) , are for customizing docker build arguments in the Dockerfile via the `ARG` directives; such as changing the base `FROM_IMAGE` between release or debug tags, adjusting the colcon mixins for each workspace to enable code coverage, or disabling fail on build failure, preventing source build breakages from blocking CI image tag updates. This also allows the slower debug CI to build and test only using the default RMW, while allowing the faster release CI jobs to build and test from an image with more RMWs installed.
+
+## Autobuild
+
+While the free registry for hosting is great, allowing projects gigabytes of free storage/bandwidth to cache pre-configured images for various branches or CI scenarios, that would otherwise take CI instances far longer to rebuild from scratch rather than pull from the registry, the automated build integration for DockerHub is basic. Rather than trigging build rules via pushed commits to matching GitHub branches, Autobuild is left disabled and the Build Trigger API is used instead. This Build Trigger API is evoked from scheduled cron jobs to rate limit DockerHub rebuilds; e.g. preventing hourly merge commits from needlessly churning CI image tags.
+
+### Build Trigger API
+
+Rather than configuring the build rules on DockerHub, a build trigger URL can be generated for the linked repo and used to programmatically specify build parameters such as: tag names, context path, source branch or version, etc. More info on Build Trigger can be found here:
+
+* [Remote Build Triggers](https://github.com/docker/docker.github.io/blob/v17.06-release/docker-hub/builds.md#remote-build-triggers)
+  * Legacy docs on using build triggers
+* [Example GitHub Action](https://github.com/osrf/docker_images/blob/master/.github/workflows/trigger_nightly.yaml)
+  * Scheduled cron job used to rebuild a nightly image
+* [Example Build Hook](https://github.com/osrf/docker_images/blob/master/ros2/nightly/nightly/hooks/post_push)
+  * A hook to rebuild a child image post push or parent tag
+
+However, triggering builds via the API rather than relying on the DockerHub build rules also means forgoing the convenience of repository linking, where a repo's image can be sure to be rebuilt as soon as a new version of the parent image tag is pushed to the registry without needing to monitor the parent image repos oneself. This helps keep the CI environment uptodate and in sync with upstream development. However, for finer control flow when triggering DockerHub build rules, this API can be called from any scheduled cron job to periodically update images when a project is least active.
+
+## Alternatives
```

Comment:
Can this go into a "Future Work" separate md file?

Reply:
Not really. The alternatives are listed for completeness, and should be listed in the same design document they are relevant to. These also aren't action items we'd necessarily seek to use for the future of nav2, just alternative options readers interested in this section should be made aware of.

Reply:
OK, can we change the title then to "Other Considered Alternatives" to make it clear that these are not in use and aren't probably going to be put in use?